src/lib.rs

```diff
@@ -142,14 +139,14 @@ impl Message {
 /// See the `tokio-core::io::Codec` Trait on how to use as a transport
 pub struct OpcCodec;
 
-impl Codec for OpcCodec {
-    type In = Message;
-    type Out = Message;
+impl Decoder for OpcCodec {
+    type Item = Message;
+    type Error = io::Error;
 
-    fn decode(&mut self, buf: &mut EasyBuf) -> io::Result<Option<Self::In>> {
+    fn decode(&mut self, buf: &mut BytesMut) -> io::Result<Option<Self::Item>> {
 
         let (msg, length) = {
-            let buf = buf.as_slice();
+            let buf = &buf[..];
```

Comment:
To be idiomatic, please use the [`Buf`](https://docs.rs/bytes/0.4.4/bytes/trait.Buf.html) trait for accessing data.  Example:
```rust
// Check if src length is more than 4;
if src.len() < 4 {
    return Err(io::Error::new(io::ErrorKind::InvalidData, "Invalid Message Command"));
}
// Create `Buf` Instance from 4 bytes
let buf = src.split_to(4).into_buf();

// Get data
let channel = buf.get_u8();
let command = buf.get_u8();
let length = buf.get_u16::<BigEndian>() as usize;

// `buf` here is now empty, repeat process. Reuse `buf` variable name by shadowing.
...

```